sdk/eventhub/Azure.Messaging.EventHubs/src/Primitives/PartitionReceiver.cs

```diff
@@ -293,6 +347,14 @@ public virtual async Task CloseAsync(CancellationToken cancellationToken = defau
             {
                 Logger.ClientCloseComplete(typeof(PartitionReceiver), EventHubName, clientHash);
             }
+
+            // If there was an active exception pending from closing the transport
+            // consumer, surface it now.
+
+            if (transportConsumerException != default)
+            {
+                throw transportConsumerException;
```

Comment:
We'll want to do something like:

```csharp
ExceptionDispatchInfo.Capture(transportConsumerException).Throw();
```

That's my bad.  Until it was mentioned in another PR, I had completely forgotten about the exception dispatcher and we left it out of scenarios like this.

Reply:
For the rationale here - it allows us to retain the stack information from the `transportConsumerException` when we caught it instead of throwing it all away when we rethrow the exception.

Reply:
Yeah.... what Matt said.  :smile: